plutus-conformance/plutus-conformance.cabal

```diff
@@ -0,0 +1,134 @@
+version: 1.0.0.0
+license: Apache-2.0
+license-files:
+  LICENSE
+  NOTICE
+maintainer: marty.stumpf@iohk.io
+author: Plutus Core Team
+synopsis: Conformance Test Suite for Plutus Core
+description:
+    Comprehensive Conformance Test Suite for Plutus Core.
+category: Language, Plutus, Conformance
+build-type: Simple
+extra-doc-files: README.md
+extra-source-files:
+    cost-model/data/builtinCostModel.json
+    cost-model/data/cekMachineCosts.json
+    cost-model/data/benching.csv
+    cost-model/data/*.R
+    plutus-core/test/CostModelInterface/defaultCostModelParams.json
+
+source-repository head
+    type: git
+    location: https://github.com/input-output-hk/plutus
+
+common lang
+    default-language: Haskell2010
+    default-extensions: ExplicitForAll FlexibleContexts ScopedTypeVariables
+                        DeriveGeneric StandaloneDeriving DeriveLift
+                        GeneralizedNewtypeDeriving DeriveFunctor DeriveFoldable
+                        DeriveTraversable DerivingStrategies DerivingVia
+                        ImportQualifiedPost
+    ghc-options: -Wall -Wnoncanonical-monad-instances
+                 -Wincomplete-uni-patterns -Wincomplete-record-updates
+                 -Wredundant-constraints -Widentities -Wunused-packages
+                 -Wmissing-deriving-strategies
+
+library
+    import: lang
+    exposed-modules:
+    hs-source-dirs:
+        uplc/
+    other-modules:
+    build-depends:
+        aeson -any,
+        algebraic-graphs >= 0.3,
+        array -any,
+        barbies -any,
+        base >=4.9 && <5,
+        bimap -any,
+        bytestring -any,
+        cardano-crypto,
+        cassava -any,
+        cborg -any,
+        composition-prelude >=1.1.0.1,
+        containers -any,
+        cryptonite -any,
+        data-default-class -any,
+        deepseq -any,
+        dependent-sum-template -any,
+        deriving-aeson >= 0.2.3,
+        deriving-compat -any,
+        dlist -any,
+        dom-lt -any,
+        exceptions -any,
+        extra -any,
+        filepath -any,
+        flat -any,
+        ghc-prim -any,
+        hashable,
+        hedgehog >=1.0,
+        int-cast -any,
+        integer-gmp -any,
+        lens -any,
+        megaparsec -any,
+        mmorph -any,
+        monoidal-containers,
+        mtl -any,
+        parser-combinators >= 0.4.0,
+        plutus-core,
+        prettyprinter >=1.1.0.1,
+        prettyprinter-configurable -any,
+        primitive -any,
+        recursion-schemes -any,
+        semigroupoids -any,
+        semigroups >=0.19.1,
+        serialise -any,
+        some < 1.0.3,
+        template-haskell -any,
+        th-compat -any,
+        text -any,
+        th-lift -any,
+        th-lift-instances -any,
+        th-utilities -any,
+        time -any,
+        transformers -any,
+        unordered-containers -any,
+        witherable -any,
+        word-array -any,
+        cardano-crypto-class -any,
+
+test-suite uplc-eval-test
```

Comment:
Given that we're going to want to instantiate the test suites multiple times, I think probably what we want is to use the *library* component to store all the test functions and then have very minimal test suites that just call the library functions with appropriate arguments, e.g.

```
-- Lib.hs
uplcEvaluationTests :: EvaluationFunction -> TestTree

-- Spec.hs
import Lib

main = defaultMain (uplcEvaluationTests haskellCekEvaluationFunction)
```